### DISCUSSION TITLE
action编译求助

### DISCUSSION THREAD

**Discussion by he110777** [Boos4721/openwrt#115] [Q&A/答疑解惑]
5.15编译不成功，请看一下是不是下面的代码有问题，5.10编译能成功的。

cd openwrt
  echo -e "$(nproc) thread compile"
  make -j$(nproc) || make -j1 || make -j1 V=s
  echo "::set-output name=status::success"
  grep '^CONFIG_TARGET.*DEVICE.*=y' .config | sed -r 's/.*DEVICE_(.*)=y/\1/' > DEVICE_NAME
  [ -s DEVICE_NAME ] && echo "DEVICE_NAME=_$(cat DEVICE_NAME)" >> $GITHUB_ENV
  echo "FILE_DATE=_$(date +"%Y%m%d%H%M")" >> $GITHUB_ENV
  shell: /usr/bin/bash -e {0}
  env:
    REPO_URL: https://github.com/Boos4721/openwrt
    REPO_BRANCH: master
    FEEDS_CONF: feeds.conf.default
    CONFIG_FILE: AX[3](https://github.com/jingleijack/AX6-AX3600_Almighty-Edition_Config/runs/6041323982?check_suite_focus=true#step:11:3)600.config
    DIY_P1_SH: diy-part1.sh
    DIY_P2_SH: diy-part2.sh
    UPLOAD_BIN_DIR: true
    UPLOAD_FIRMWARE: true
    UPLOAD_COWTRANSFER: false
    UPLOAD_WETRANSFER: false
    UPLOAD_RELEASE: true
    TZ: Asia/Shanghai
2 thread compile
 make[1] world
 make[2] tools/compile
 make[2] package/cleanup
 make[3] -C tools/flock compile
 make[3] -C tools/xz compile
 make[3] -C tools/sed compile
 make[3] -C tools/tar compile
 make[3] -C tools/patch compile
 make[3] -C tools/pkgconf compile
 make[3] -C tools/ninja compile
 make[3] -C tools/libressl compile
 make[3] -C tools/meson compile
 make[3] -C tools/zstd compile
 make[3] -C tools/cmake compile
 make[3] -C tools/ccache compile
 make[3] -C tools/m[4](https://github.com/jingleijack/AX6-AX3600_Almighty-Edition_Config/runs/6041323982?check_suite_focus=true#step:11:4) compile
 make[3] -C tools/autoconf-archive compile
 make[3] -C tools/cpio compile
 make[3] -C tools/zlib compile
 make[3] -C tools/lzma compile
 make[3] -C tools/mtools compile
 make[3] -C tools/patch-image compile
 make[3] -C tools/squashfskit4 compile
 make[3] -C tools/sstrip compile
 make[3] -C tools/ucl compile
 make[3] -C tools/xxd compile
 make[3] -C tools/zip compile
  HOSTCC  tools/libfdt/fdt_sw.o
  HOSTCC  tools/libfdt/fdt_rw.o
  HOSTCC  tools/libfdt/fdt_strerror.o
  HOSTCC  tools/libfdt/fdt_empty_tree.o
  HOSTCC  tools/libfdt/fdt_addresses.o
  HOSTCC  tools/libfdt/fdt_overlay.o
  HOSTCC  tools/gpimage.o
  HOSTCC  tools/gpimage-common.o
  HOSTCC  tools/mtk_image.o
  HOSTCC  tools/lib/ecdsa/ecdsa-libcrypto.o
  HOSTCC  tools/lib/rsa/rsa-sign.o
  HOSTCC  tools/lib/rsa/rsa-verify.o
  HOSTCC  tools/lib/rsa/rsa-mod-exp.o
  HOSTCC  tools/lib/aes/aes-encrypt.o
  HOSTCC  tools/lib/aes/aes-decrypt.o
  HOSTCC  tools/dumpimage.o
  HOSTLD  tools/dumpimage
/usr/bin/ld: /workdir/openwrt/staging_dir/host/lib/libssl.a(ssl_init.o): in function `OPENSSL_init_ssl':
ssl_init.c:(.text+0x71): undefined reference to `pthread_once'
/usr/bin/ld: /workdir/openwrt/staging_dir/host/lib/libcrypto.a(libcrypto_la-crypto_init.o): in function `OPENSSL_init_crypto':
crypto_init.c:(.text+0x[5](https://github.com/jingleijack/AX6-AX3600_Almighty-Edition_Config/runs/6041323982?check_suite_focus=true#step:11:5)5): undefined reference to `pthread_once'
/usr/bin/ld: /workdir/openwrt/staging_dir/host/lib/libcrypto.a(libcrypto_la-conf_sap.o): in function `OpenSSL_config':
conf_sap.c:(.text+0xc0): undefined reference to `pthread_once'
/usr/bin/ld: /workdir/openwrt/staging_dir/host/lib/libcrypto.a(libcrypto_la-conf_sap.o): in function `OpenSSL_no_config':
conf_sap.c:(.text+0x107): undefined reference to `pthread_once'
/usr/bin/ld: /workdir/openwrt/staging_dir/host/lib/libcrypto.a(libcrypto_la-eng_all.o): in function `ENGINE_load_builtin_engines':
eng_all.c:(.text+0x34): undefined reference to `pthread_once'
/usr/bin/ld: /workdir/openwrt/staging_dir/host/lib/libcrypto.a(libcrypto_la-err.o):err.c:(.text+0xad2): more undefined references to `pthread_once' follow
collect2: error: ld returned 1 exit status
make[[6](https://github.com/jingleijack/AX6-AX3600_Almighty-Edition_Config/runs/6041323982?check_suite_focus=true#step:11:6)]: *** [scripts/Makefile.host:104: tools/dumpimage] Error 1
make[5]: *** [Makefile:2081: tools-only] Error 2
make[4]: *** [Makefile:555: __build_one_by_one] Error 2
make[4]: Leaving directory '/workdir/openwrt/build_dir/host/u-boot-2022.01'
make[3]: *** [Makefile:44: /workdir/openwrt/build_dir/host/u-boot-2022.01/.built] Error 2
make[3]: Leaving directory '/workdir/openwrt/tools/mkimage'
time: tools/mkimage/compile#11.68#2.11#13.[7](https://github.com/jingleijack/AX6-AX3600_Almighty-Edition_Config/runs/6041323982?check_suite_focus=true#step:11:7)5
    ERROR: tools/mkimage failed to build.
make[2]: *** [tools/Makefile:170: tools/mkimage/compile] Error 1
make[2]: Leaving directory '/workdir/openwrt'
make[1]: *** [tools/Makefile:[16](https://github.com/jingleijack/AX6-AX3600_Almighty-Edition_Config/runs/6041323982?check_suite_focus=true#step:11:16)6: /workdir/openwrt/staging_dir/host/stamp/.tools_compile_yyynyynnyyyyynyyyyyynyynnyynynyyyyyyyyyyyyyyyyyynynnyyyyyyyyyy] Error 2
make[1]: Leaving directory '/workdir/openwrt'
make: *** [/workdir/openwrt/include/toplevel.mk:[23](https://github.com/jingleijack/AX6-AX3600_Almighty-Edition_Config/runs/6041323982?check_suite_focus=true#step:11:23)0: world] Error 2
Error: Process completed with exit code 2.

**Comment by Boos4721** [comment]
EVERYTHING WORKS
